train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -13,6 +13,16 @@ public class TrainControllerImpl implements TrainController {
 	private int speedLimit = 0;
 	private boolean speedBoostActive;
 
+	public TrainControllerImpl() {
+		while(true) {
+			try {
+				followSpeed();
```

Comment:
this should be inside a new thread to not block the main thread and that would let the tests complete